File: Python/day01/day01.py
with open('day01.txt', 'r') as f :
    inventories = f.read().split("\n\n")

caloriesPerElf = []
for inventory in inventories :
    inventoryContents = inventory.strip().split("\n")
    calories = sum( int(item) for item in inventoryContents )
    caloriesPerElf.append(calories)

print(f"Part 1. {max(caloriesPerElf)}")
print(f"Part 2. {sum(sorted(caloriesPerElf)[-3:])}")

File: src/main/kotlin/day01/Day01.kt
package day01

import java.io.File

val elfInventories = File("src/main/resources/day01.txt").readText().trim().split("\n\n")

fun main() {
    val caloriesPerElf: List<Int> = elfInventories.map { inventory ->
        inventory.lines().sumOf { it.toInt() }
    }

    val highestCaloriesInventory = caloriesPerElf.max()

    val sumOfThreeHighest = caloriesPerElf.sorted().takeLast(3).sum()

    println("Part 1. $highestCaloriesInventory")
    println("Part 2. $sumOfThreeHighest")
}